client/actions/assignments/api.ts

```diff
@@ -469,12 +473,55 @@ const receiveAssignmentHistory = (items) => ({
  * Fetch the Event and/or Planning item(s) associated with this Assignment
  * @param {object} assignment - The Assignment to load items for
  */
-const loadPlanningAndEvent = (assignment) => (
-    (dispatch) => (
-        dispatch(planning.api.fetchById(assignment.planning_item))
-    )
-);
+const loadPlanningAndEvent = (assignment) => (dispatch) =>
+    dispatch(planning.api.fetchById(assignment.planning_item));
+
+/**
+ * Loads the Archive items that are linked to the provided Assignment list
+ * The Archive items is then saved to the redux store under
+ * assignment.archive dictionary
+ */
+const loadArchiveItems = (assignments: Array<any>) => (
+    dispatch,
+    getState,
+    {api, notify, search}
+) => {
+    const assignmentIds = assignments.map((a) => a._id);
+
+    const query = search.query();
+    const filter = {
+        bool: {
+            should: assignmentIds.map((id) => ({term: {assignment_id: id}})),
```

Comment:
Re-fetching all assignments on `content:update` might be a bit much. It might be better to only reload the content items that are changed based on the ids provided in the `content:update` notification.

`should` can also be changed to a `must.terms` query, i.e.
`must: {terms: {_id: content_ids}}`